presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -849,4 +862,65 @@ private static String typedNull(String prestoType)
     {
         return format("CAST(NULL AS %s)", prestoType);
     }
+
+    private static List<ColumnInfo> getDescribeOutputColumnInfoList(ResultSet resultSet)
+            throws SQLException
+    {
+        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
+        while (resultSet.next()) {
+            String columnName = resultSet.getString("Column Name");
+            String catalog = resultSet.getString("Catalog");
+            String schema = resultSet.getString("Schema");
+            String table = resultSet.getString("Table");
+            ClientTypeSignature clientTypeSignature = getClientTypeSignatureFromTypeString(resultSet.getString("Type"));
+            ColumnInfo.Builder builder = new ColumnInfo.Builder()
+                    .setColumnName(columnName)
+                    .setColumnLabel(columnName)
+                    .setCatalogName(catalog)
+                    .setSchemaName(schema)
+                    .setTableName(table)
+                    .setColumnTypeSignature(clientTypeSignature)
+                    .setNullable(ColumnInfo.Nullable.UNKNOWN);
+            setTypeInfo(builder, clientTypeSignature);
+            list.add(builder.build());
+        }
+        return list.build();
+    }
+
+    @VisibleForTesting
+    static ClientTypeSignature getClientTypeSignatureFromTypeString(String type)
+    {
+        Matcher topLevelMatcher = topLevelTypeRegexPattern.matcher(type);
+        String topLevelType;
+        List<ClientTypeSignatureParameter> arguments = new ArrayList<>();
+        if (topLevelMatcher.find()) {
+            topLevelType = topLevelMatcher.group(1);
+            String typeParameters = topLevelMatcher.group(2);
+            if (topLevelType.equals("decimal")) {
+                String[] precisionAndScale = typeParameters.split(",");
+                long precision = Long.parseLong(precisionAndScale[0]);
+                arguments.add(ClientTypeSignatureParameter.ofLong(precision));
+                long scale = Long.parseLong(precisionAndScale[1]);
+                arguments.add(ClientTypeSignatureParameter.ofLong(scale));
+            }
+            else if (topLevelType.equals("char") || topLevelType.equals("varchar")) {
+                long precision = Long.parseLong(typeParameters);
+                arguments.add(ClientTypeSignatureParameter.ofLong(precision));
+            }
+        }
+        else {
+            Matcher timestampTimeZonePrecisionMatcher = timestampTimeZonePrecisionRegexPattern.matcher(type);
+            if (timestampTimeZonePrecisionMatcher.find()) {
+                topLevelType = "timestamp with time zone";
+                arguments.add(ClientTypeSignatureParameter.ofLong(Long.parseLong(timestampTimeZonePrecisionMatcher.group(1))));
+            }
+            else {
+                topLevelType = type;
+                if (topLevelType.equals("varchar")) {
+                    arguments.add(ClientTypeSignatureParameter.ofLong(Integer.MAX_VALUE));
```

Comment:
This should be `Integer.MAX_VALUE - 1` to match `io.prestosql.spi.type.VarcharType#MAX_LENGTH`

We might want to add a `VARCHAR_MAX_LENGTH` constant in this class, for use here and in the tests